draft-ietf-tls-esni.md

```diff
@@ -416,9 +416,21 @@ OuterExtensions consists of one or more ExtensionType values, each of which
 reference an extension in ClientHelloOuter.
 
 When sending ClientHello, the client first computes ClientHelloInner, including
-any PSK binders. It then computes a new value, the EncodedClientHelloInner, by
-first making a copy of ClientHelloInner. It then replaces the
-legacy\_session\_id field with an empty string.
+any PSK binders. It then computes a new value, the EncodedClientHelloInner,
+which is the following structure:
+
+~~~
+    struct {
+        ClientHello client_hello;
+        opaque padding<0..2^16-1>;
```

Comment:
This would be easier if this wasn't length-prefixed.  Every padding scheme that ever used length has caused a lot of pain.  I realize that this isn't necessarily security-critical, but just extending with zeros should be easier to implement AND you don't have two extra bytes of overhead.

Reply:
Works for me. Since every TLS structure is self-delimiting, we can just append stuff. It shifts some work from presentation language to prose, but I agree that it's easier to implement.

Reply:
Done. I kinda cheated a bit around the presentation language, but hopefully that's clear enough?